Data/Memory/Internal/CompatPrim64.hs

```diff
@@ -63,6 +63,7 @@ type OutBool = Bool
 type Word64# = Word#
 type Int64# = Int#
 
+#if __GLASGOW_HASKELL__ < 904
```

Comment:
Relevant note in migration guide: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.4#word64int64-use-word64int64

Reply:
this conditional seems rather suspicious for a 9.4 compat change. Is it correct?

Reply:
The `904` here means 9.4, not 9.0.4 as you might expect. See the documentation for a more thorough explanation: https://downloads.haskell.org/~ghc/9.4.2/docs/users_guide/phases.html#standard-cpp-macros